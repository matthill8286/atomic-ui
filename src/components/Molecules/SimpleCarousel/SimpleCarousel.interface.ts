import Slider, { Settings } from 'react-slick'

import { Product, ThemeColors } from '@/types'
import { BadgeType } from '@/components/Atoms/Badge'

export interface SliderItemProps {
  headline?: string
  usageType?: string
  link?: string | null
  metadata?: string | null
  productID?: string
  sortOrder?: number
  title?: string
  url?: string
}

// eslint-disable-next-line @typescript-eslint/no-explicit-any
type Item = SliderItemProps | Product | any
export interface CarouselSettings extends Settings {
  key?: string
}

export interface CarouselProps {
  sliderSettings: CarouselSettings
  items: Item[]
  badges?: BadgeType[]
  whiteDots?: boolean
  brightArrows?: boolean
  boxShadow?: boolean
  hasOverflow?: boolean
  fixedWidth?: boolean
  tileWidth?: number
  hasPadding?: boolean
  renderSlide(entry: Item, index: number): JSX.Element | null
  className?: string
  carouselRef?: (e: Slider) => void
  arrowsSurfaceColor?: ThemeColors
}

export interface StyledArrowProps {
  showArrows: boolean
}
