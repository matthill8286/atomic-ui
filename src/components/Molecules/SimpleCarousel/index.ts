export * from './SimpleCarousel'
export * from './SimpleCarousel.interface'
