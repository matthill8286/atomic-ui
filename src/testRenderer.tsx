import { mount, ReactWrapper } from 'enzyme'
import React from 'react'
import { MemoryRouter as Router } from 'react-router-dom'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import { saiyanTheme } from '@/styles/sc-vars-saiyan'
import { Theme } from '@/types'

export function renderWithTheme<C extends React.Component, P = C['props'], S = C['state']>(
  tree: React.ReactElement<P>,
  theme = saiyanTheme
) {
  return renderer.create(<ThemeProvider theme={theme}>{tree}</ThemeProvider>).toJSON()
}

export function renderWithThemeAndRouter(tree: React.ReactElement, theme: Theme = saiyanTheme) {
  return renderer
    .create(
      <ThemeProvider theme={theme}>
        <Router>{tree}</Router>
      </ThemeProvider>
    )
    .toJSON()
}

export function renderWithRouter(tree: React.ReactElement) {
  return renderer.create(<Router>{tree}</Router>).toJSON()
}

export function mountWithTheme<C extends React.Component, P = C['props'], S = C['state']>(
  tree: React.ReactElement<P>
): ReactWrapper<P, S, C> {
  return mount(<ThemeProvider theme={saiyanTheme}>{tree}</ThemeProvider>)
}

export function mountWithThemeAndRouter<C extends React.Component, P = C['props'], S = C['state']>(
  tree: React.ReactElement<P>
): ReactWrapper<P, S, C> {
  return mount(
    <ThemeProvider theme={saiyanTheme}>
      <Router>{tree}</Router>
    </ThemeProvider>
  )
}
