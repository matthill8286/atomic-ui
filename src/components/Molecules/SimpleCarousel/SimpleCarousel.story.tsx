import { storiesOf } from '@storybook/react'
import React from 'react'
import { Badge } from '@/components/Atoms/Badge'
import { Picture } from '@/components/Atoms/Picture'
import { VideoPlayerCore } from '@/components/Atoms/VideoPlayer'
import { ThemeProvider } from '@/styles/ThemeProvider'
import { SimpleCarousel } from './SimpleCarousel'
import {
  defaultSettings,
  multiItemSettings,
  pictureEntries,
  withVideoEntries,
} from './SimpleCarousel.data'
import { saiyanTheme } from '@/styles'

const carouselWrapperStyle = {
  maxWidth: '634px',
  margin: '0 auto',
}

const badges = [
  {
    id: 1,
    name: 'Sponsored',
  },
  {
    id: 2,
    name: 'Hot Hot Hot',
  },
]

const blackExampleTheme = {
  ...saiyanTheme,
  color: {
    ...saiyanTheme.color,
    contrastColor: '#454647',
  },
}

export const renderWithPictureComponent = (item, index) => {
  return (
    <div key={item.productID + index}>
      <Picture src={item.url} alt={item.productID} />
    </div>
  )
}

const renderWithPictureOrVideoComponent = (item, index) => {
  if (item.usageType === 'Product Video') {
    return <VideoPlayerCore key={item.videoID} videoId={item.videoID} url={item.url} />
  } else {
    return renderWithPictureComponent(item, index)
  }
}

storiesOf('Design System/Molecules/Simple Carousel', module)
  .add('Default', () => (
    <div style={carouselWrapperStyle}>
      <SimpleCarousel
        sliderSettings={defaultSettings}
        items={pictureEntries}
        renderSlide={(item, index) => renderWithPictureComponent(item, index)}
      />
    </div>
  ))
  .add('with MultiItem', () => (
    <div style={carouselWrapperStyle}>
      <SimpleCarousel
        sliderSettings={multiItemSettings}
        items={pictureEntries}
        renderSlide={(item, index) => renderWithPictureComponent(item, index)}
      />
    </div>
  ))
  .add('with Video', () => (
    <div style={carouselWrapperStyle}>
      <SimpleCarousel
        boxShadow
        sliderSettings={defaultSettings}
        items={withVideoEntries}
        renderSlide={(item, index) => renderWithPictureOrVideoComponent(item, index)}
      />
    </div>
  ))
  .add('with Badges', () => (
    <div style={carouselWrapperStyle}>
      <Badge badges={badges} />
      <SimpleCarousel
        boxShadow
        sliderSettings={defaultSettings}
        items={withVideoEntries}
        renderSlide={(item, index) => renderWithPictureOrVideoComponent(item, index)}
      />
    </div>
  ))
  .add('With black Alternate theme', () => (
    <div style={{ backgroundColor: 'black' }}>
      <div style={carouselWrapperStyle}>
        <ThemeProvider theme={blackExampleTheme}>
          <SimpleCarousel
            sliderSettings={defaultSettings}
            items={pictureEntries}
            renderSlide={(item, index) => renderWithPictureComponent(item, index)}
            whiteDots
            arrowsSurfaceColor="contrastColor"
          />
        </ThemeProvider>
      </div>
    </div>
  ))
