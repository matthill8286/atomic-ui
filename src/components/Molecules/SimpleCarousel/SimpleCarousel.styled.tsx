import { css, styled } from '@/styles/styled'
import { CarouselProps } from './SimpleCarousel.interface'

const StyledArrow = css<CarouselProps>`
  ${({ theme, arrowsSurfaceColor, brightArrows = false }) => css`
    background: ${brightArrows ? theme.color.white : theme.color.black};

    display: none !important;
    :hover {
      display: block !important;
    }
    & svg {
      fill: ${brightArrows ? theme.color.black : theme.color.white} !important;
    }

    ${arrowsSurfaceColor &&
      css`
        background-color: ${theme.color[arrowsSurfaceColor]};
      `}
  `}
`

export const StyledCarouselHandler = styled.div`
  outline: none;
`

export const StyledCarousel = styled.div<CarouselProps>`
  width: 100%;
  position: relative;
  margin: 0 auto;

  ${({ theme, boxShadow }) =>
    boxShadow &&
    css`
      & .slick-list {
        box-shadow: ${theme.dimension.elevationLevel2};
      }
    `}

  & .slick-slider {
    position: relative;
    display: block;
    box-sizing: border-box;
    user-select: none;
    touch-action: pan-y;
    -webkit-tap-highlight-color: transparent;

    & .slick-list {
      position: relative;
      padding: ${({ theme }) => `0 0 ${theme.spacing.base.sm} 0`};
      transform: translate3d(0, 0, 0);
      box-sizing: content-box;
      overflow: ${({ hasOverflow }) => (hasOverflow ? 'visible' : 'hidden')};
      margin: ${({ hasPadding }) => (hasPadding ? '0 -12px' : '0')};

      & .slick-track {
        display: flex;
        justify-content: center;
        position: relative;
        top: 0;
        left: 0;
        transform: translate3d(0, 0, 0);
        margin-left: 0;
        margin-right: auto;

        & .slick-slide {
          display: flex;
          align-items: center;
          justify-content: center;
          -webkit-justify-content: center;
          height: auto;
          position: relative;
          float: left;
          ${({ hasPadding }) => hasPadding && 'margin: 0 12px'};
          ${({ fixedWidth, tileWidth }) => fixedWidth && `min-width: ${tileWidth}px !important`};

          & img {
            display: block;
            margin: auto;
            max-width: 100%;
          }

          & > div {
            width: ${({ fixedWidth, tileWidth }) =>
              fixedWidth ? `${tileWidth}px` ?? '100%' : '100%'};
            height: 100%;
            display: flex;
          }
        }
      }
    }

    :hover {
      & .slick-next,
      & .slick-prev {
        display: block !important;
      }
      & .slick-next {
        right: -1%;
      }
      & .slick-prev {
        left: -1%;
      }
    }

    & .slick-dots {
      position: absolute;
      top: 0;
      right: ${({ theme }) => theme.spacing.base.xs};
      max-height: ${({ theme }) => theme.spacing.base.sm};
      width: fit-content;
      list-style: none;
      display: block;
      text-align: center;
      margin: ${({ theme }) => `-${theme.spacing.base.sm} auto 0 auto`};

      & li {
        position: relative;
        display: inline-block;
        padding: 0;
        margin-right: ${({ theme }) => theme.spacing.base.sm};

        & div {
          background-color: transparent;
          border-color: ${({ theme }) => theme.color.primary};
          border-radius: ${({ theme }) => theme.spacing.base.xs};
          display: block;
          height: ${({ theme }) => theme.spacing.base.xs};
          width: ${({ theme }) => theme.spacing.base.xs};
          outline: none;
          color: transparent;
          border-style: solid;
          border-width: 1px;
          box-sizing: border-box;
          &:hover {
            cursor: pointer;
          }
        }

        &.slick-active div {
          background: ${({ theme }) => theme.color.primary};
          opacity: 1;
        }
      }
    }
  }
  & .slick-arrow {
    /* reset the default behaviour */
    ::before {
      content: unset;
    }
    z-index: 1;
    position: absolute;
    top: 50%;
    outline: 0;
    transform: translateY(
      calc(-50% - 19px)
    ); /* -19px because of the additional space that is needed for the slick-slider dots */
    border: none;

    & svg {
      position: relative;
    }

    &:hover {
      cursor: pointer;
    }

    &.slick-disabled {
      opacity: 0;
      &:hover {
        cursor: default;
      }
    }

    &.slick-next {
      ${StyledArrow};
      right: 3%;
      border-radius: ${({ theme }) => theme.dimension.borderRadius5};
    }

    &.slick-prev {
      ${StyledArrow};
      left: 3%;
      border-radius: ${({ theme }) => theme.dimension.borderRadius5};
    }
  }

  ${({ whiteDots, theme }) =>
    whiteDots &&
    css`
      & .slick-dots {
        top: 85%;
        padding: 0 !important;
      }
      & .slick-dots > li > div {
        border-color: ${theme.color.white} !important;
        background-color: transparent;
      }
      & .slick-dots .slick-active > div {
        background-color: ${theme.color.white} !important;
        border-color: ${theme.color.white} !important;
      }
    `}
`
export const StyledNavArrow = styled.div`
  position: absolute;
  background: ${({ theme }) => theme.color.white};
  fill: ${({ theme }) => theme.color.grey6};
  border-radius: ${({ theme }) => theme.dimension.borderRadius7};
  box-shadow: ${({ theme }) => theme.dimension.elevationLevel3};
  height: ${({ theme }) => theme.spacing.base.xxxl};
  width: ${({ theme }) => theme.spacing.base.xxxl};

  cursor: pointer;
  z-index: 3;
  display: none;
`
