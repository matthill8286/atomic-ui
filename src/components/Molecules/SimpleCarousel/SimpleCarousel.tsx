import React, { useCallback, useState } from 'react'
import Slider from 'react-slick'
import { Icon, RotationValues } from '@/components/Atoms/Icon'
import { CarouselProps } from './SimpleCarousel.interface'
import { StyledCarousel, StyledCarouselHandler, StyledNavArrow } from './SimpleCarousel.styled'
import { StyleguideArrow } from '@matthill8286/atomic-icon-library'

export const SimpleCarousel: React.FC<CarouselProps> = ({
  arrowsSurfaceColor,
  sliderSettings,
  items,
  renderSlide,
  whiteDots,
  brightArrows,
  boxShadow,
  hasOverflow,
  hasPadding,
  fixedWidth,
  tileWidth,
  className,
  carouselRef,
}): React.ReactElement => {
  let firstClientX: number | 0

  const sliderRef = React.useRef<Slider | null>(null)
  const minValue = 20 // threshold

  const rightDirection = (direction: number) => direction < -Math.abs(minValue)
  const leftDirection = (direction: number) => direction > Math.abs(minValue)

  const touchStart = (e: React.TouchEvent): void => {
    e.preventDefault()
    e.stopPropagation()
    firstClientX = e.touches[0].clientX
  }

  const touchEnd = (e: React.TouchEvent): void => {
    const clientX = Math.round(e.changedTouches[0].clientX - firstClientX)

    if (rightDirection(clientX)) {
      sliderRef.current?.slickNext()
    } else if (leftDirection(clientX)) {
      sliderRef.current?.slickPrev()
    }
  }

  const renderArrows = (rotation: RotationValues): React.ReactElement => {
    return (
      <StyledNavArrow>
        <Icon rotate={rotation} color="white" padding="sm">
          <StyleguideArrow width={30} height={30} />
        </Icon>
      </StyledNavArrow>
    )
  }

  const customPaging = () => <div />

  const sliderConfig = () => {
    return {
      focusOnSelect: false,
      nextArrow: renderArrows(0),
      prevArrow: renderArrows(180),
      customPaging,
      responsive: [
        {
          breakpoint: 1007,
          settings: {
            swipe: false,
            swipeToSlide: false,
            draggable: false,
            touchMove: false,
          },
        },
        {
          breakpoint: 1008,
          settings: {
            swipeToSlide: true,
          },
        },
      ],
      ...sliderSettings,
      ref: e => {
        sliderRef.current = e
        if (carouselRef) {
          carouselRef(e)
        }
      },
    }
  }

  const [dragging, setDragging] = useState(false)

  const handleBeforeChange = useCallback(() => {
    setDragging(true)
  }, [setDragging])

  const handleAfterChange = useCallback(() => {
    setDragging(false)
  }, [setDragging])

  const handleOnItemClick = useCallback(
    e => {
      if (dragging) {
        e.stopPropagation()
      }
    },
    [dragging]
  )

  const beforeChange = (currentSlide, nextSlide) => {
    if (sliderSettings.beforeChange) sliderSettings.beforeChange(currentSlide, nextSlide)
    handleBeforeChange()
  }

  const afterChange = currentSlide => {
    if (sliderSettings.afterChange) sliderSettings.afterChange(currentSlide)
    handleAfterChange()
  }

  return (
    <StyledCarousel
      arrowsSurfaceColor={arrowsSurfaceColor}
      whiteDots={whiteDots}
      fixedWidth={fixedWidth}
      tileWidth={tileWidth}
      brightArrows={brightArrows}
      items={items}
      sliderSettings={sliderSettings}
      renderSlide={renderSlide}
      boxShadow={boxShadow}
      hasOverflow={hasOverflow}
      hasPadding={hasPadding}
      className={className}
      onTouchStart={touchStart}
      onTouchEnd={touchEnd}>
      <Slider {...sliderConfig()} beforeChange={beforeChange} afterChange={afterChange}>
        {items.map((entry, index) => {
          return (
            <StyledCarouselHandler
              onClickCapture={handleOnItemClick}
              key={index}
              className="ewb-slider--element">
              {renderSlide(entry, index)}
            </StyledCarouselHandler>
          )
        })}
      </Slider>
    </StyledCarousel>
  )
}

SimpleCarousel.displayName = 'SimpleCarousel'
