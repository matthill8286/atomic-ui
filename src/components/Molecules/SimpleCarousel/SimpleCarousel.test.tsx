import React from 'react'
import { mountWithTheme } from '@/testRenderer'
import { SimpleCarousel } from './SimpleCarousel'
import { pictureEntries } from './SimpleCarousel.data'
import { renderWithPictureComponent } from './SimpleCarousel.story'

describe('SimpleCarousel', () => {
  it('renders correctly', () => {
    const tree = mountWithTheme(
      <SimpleCarousel
        items={pictureEntries}
        renderSlide={renderWithPictureComponent}
        sliderSettings={{
          dots: true,
          infinite: false,
          slidesToScroll: 1,
          slidesToShow: 1,
          speed: 500,
        }}
      />
    )
    expect(tree).toMatchSnapshot()
  })
})
