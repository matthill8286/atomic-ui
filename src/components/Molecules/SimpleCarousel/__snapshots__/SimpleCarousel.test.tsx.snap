// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleCarousel renders correctly 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 16px;
}

.c2 svg {
  width: 25px;
  height: 25px;
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
  fill: #FFFFFF;
}

.c2 svg g,
.c2 svg p {
  fill: #FFFFFF;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 16px;
}

.c5 svg {
  width: 25px;
  height: 25px;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  fill: #FFFFFF;
}

.c5 svg g,
.c5 svg p {
  fill: #FFFFFF;
}

.c3 {
  outline: none;
}

.c0 {
  width: 100%;
  position: relative;
  margin: 0 auto;
}

.c0 .slick-slider {
  position: relative;
  display: block;
  box-sizing: border-box;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  touch-action: pan-y;
  -webkit-tap-highlight-color: transparent;
}

.c0 .slick-slider .slick-list {
  position: relative;
  padding: 0 0 16px 0;
  -webkit-transform: translate3d(0,0,0);
  -ms-transform: translate3d(0,0,0);
  transform: translate3d(0,0,0);
  box-sizing: content-box;
  overflow: hidden;
  margin: 0;
}

.c0 .slick-slider .slick-list .slick-track {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  top: 0;
  left: 0;
  -webkit-transform: translate3d(0,0,0);
  -ms-transform: translate3d(0,0,0);
  transform: translate3d(0,0,0);
  margin-left: 0;
  margin-right: auto;
}

.c0 .slick-slider .slick-list .slick-track .slick-slide {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-justify-content: center;
  height: auto;
  position: relative;
  float: left;
}

.c0 .slick-slider .slick-list .slick-track .slick-slide img {
  display: block;
  margin: auto;
  max-width: 100%;
}

.c0 .slick-slider .slick-list .slick-track .slick-slide > div {
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 .slick-slider:hover .slick-next,
.c0 .slick-slider:hover .slick-prev {
  display: block !important;
}

.c0 .slick-slider:hover .slick-next {
  right: -1%;
}

.c0 .slick-slider:hover .slick-prev {
  left: -1%;
}

.c0 .slick-slider .slick-dots {
  position: absolute;
  top: 0;
  right: 8px;
  max-height: 16px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  list-style: none;
  display: block;
  text-align: center;
  margin: -16px auto 0 auto;
}

.c0 .slick-slider .slick-dots li {
  position: relative;
  display: inline-block;
  padding: 0;
  margin-right: 16px;
}

.c0 .slick-slider .slick-dots li div {
  background-color: transparent;
  border-color: #00C8DC;
  border-radius: 8px;
  display: block;
  height: 8px;
  width: 8px;
  outline: none;
  color: transparent;
  border-style: solid;
  border-width: 1px;
  box-sizing: border-box;
}

.c0 .slick-slider .slick-dots li div:hover {
  cursor: pointer;
}

.c0 .slick-slider .slick-dots li.slick-active div {
  background: #00C8DC;
  opacity: 1;
}

.c0 .slick-arrow {
  z-index: 1;
  position: absolute;
  top: 50%;
  outline: 0;
  -webkit-transform: translateY( calc(-50% - 19px) );
  -ms-transform: translateY( calc(-50% - 19px) );
  transform: translateY( calc(-50% - 19px) );
  border: none;
}

.c0 .slick-arrow::before {
  content: unset;
}

.c0 .slick-arrow svg {
  position: relative;
}

.c0 .slick-arrow:hover {
  cursor: pointer;
}

.c0 .slick-arrow.slick-disabled {
  opacity: 0;
}

.c0 .slick-arrow.slick-disabled:hover {
  cursor: default;
}

.c0 .slick-arrow.slick-next {
  background: #000000;
  display: none !important;
  right: 3%;
  border-radius: 20px;
}

.c0 .slick-arrow.slick-next:hover {
  display: block !important;
}

.c0 .slick-arrow.slick-next svg {
  fill: #FFFFFF !important;
}

.c0 .slick-arrow.slick-prev {
  background: #000000;
  display: none !important;
  left: 3%;
  border-radius: 20px;
}

.c0 .slick-arrow.slick-prev:hover {
  display: block !important;
}

.c0 .slick-arrow.slick-prev svg {
  fill: #FFFFFF !important;
}

.c1 {
  position: absolute;
  background: #FFFFFF;
  fill: #333333;
  border-radius: 40px;
  box-shadow: 0 8px 10px 1px rgba(0,0,0,0.07),0 3px 14px 2px rgba(0,0,0,0.06),0 5px 5px -3px rgba(0,0,0,0.10);
  height: 56px;
  width: 56px;
  cursor: pointer;
  z-index: 3;
  display: none;
}

.c4 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  max-width: 100%;
}

.c4 img {
  object-fit: cover;
}

@media screen and (-ms-high-contrast:active),screen and (-ms-high-contrast:none) {
  .c4 {
    max-width: 100%;
  }
}

@media screen and (-ms-high-contrast:active),screen and (-ms-high-contrast:none) {
  .c4 img {
    max-width: 100%;
    width: auto;
    height: auto;
  }
}

<div
  className="c0"
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
>
  <div
    className="slick-slider slick-initialized"
    dir="ltr"
  >
    <div
      className="c1 slick-arrow slick-prev slick-disabled"
      data-role="none"
      onClick={null}
      style={
        Object {
          "display": "block",
        }
      }
    >
      <div
        className="c2"
        color="white"
        rotate={180}
      >
        <svg
          height={30}
          viewBox="0 0 24 24"
          width={30}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.78 11.28a.75.75 0 01.22.53v.38a.77.77 0 01-.22.53l-5.14 5.13a.5.5 0 01-.71 0l-.71-.71a.49.49 0 010-.7L13.67 12 9.22 7.56a.5.5 0 010-.71l.71-.7a.5.5 0 01.71 0z"
          />
        </svg>
      </div>
    </div>
    <div
      className="slick-list"
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={null}
      onMouseMove={null}
      onMouseUp={[Function]}
      onTouchCancel={null}
      onTouchEnd={[Function]}
      onTouchMove={null}
      onTouchStart={[Function]}
      style={Object {}}
    >
      <div
        className="slick-track"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        style={
          Object {
            "WebkitTransform": "translate3d(0px, 0px, 0px)",
            "WebkitTransition": "",
            "msTransform": "translateX(0px)",
            "opacity": 1,
            "transform": "translate3d(0px, 0px, 0px)",
            "transition": "",
          }
        }
      >
        <div
          aria-hidden={false}
          className="slick-slide slick-active slick-current"
          data-index={0}
          onClick={[Function]}
          style={
            Object {
              "outline": "none",
              "width": 0,
            }
          }
          tabIndex="-1"
        >
          <div>
            <div
              className="c3 ewb-slider--element"
              onClickCapture={[Function]}
              style={
                Object {
                  "display": "inline-block",
                  "width": "100%",
                }
              }
              tabIndex={-1}
            >
              <div>
                <div
                  className="c4"
                  width="100%"
                >
                  <picture
                    style={
                      Object {
                        "height": undefined,
                        "width": "100%",
                      }
                    }
                  >
                    <img
                      alt="featured-product-69001598"
                      loading="eager"
                      src="public/images/teasers/1.jpg"
                      width="100%"
                    />
                  </picture>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          aria-hidden={true}
          className="slick-slide"
          data-index={1}
          onClick={[Function]}
          style={
            Object {
              "outline": "none",
              "width": 0,
            }
          }
          tabIndex="-1"
        >
          <div>
            <div
              className="c3 ewb-slider--element"
              onClickCapture={[Function]}
              style={
                Object {
                  "display": "inline-block",
                  "width": "100%",
                }
              }
              tabIndex={-1}
            >
              <div>
                <div
                  className="c4"
                  width="100%"
                >
                  <picture
                    style={
                      Object {
                        "height": undefined,
                        "width": "100%",
                      }
                    }
                  >
                    <img
                      alt="featured-product-69001599"
                      loading="eager"
                      src="public/images/teasers/02.jpg"
                      width="100%"
                    />
                  </picture>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          aria-hidden={true}
          className="slick-slide"
          data-index={2}
          onClick={[Function]}
          style={
            Object {
              "outline": "none",
              "width": 0,
            }
          }
          tabIndex="-1"
        >
          <div>
            <div
              className="c3 ewb-slider--element"
              onClickCapture={[Function]}
              style={
                Object {
                  "display": "inline-block",
                  "width": "100%",
                }
              }
              tabIndex={-1}
            >
              <div>
                <div
                  className="c4"
                  width="100%"
                >
                  <picture
                    style={
                      Object {
                        "height": undefined,
                        "width": "100%",
                      }
                    }
                  >
                    <img
                      alt="featured-product-69001597"
                      loading="eager"
                      src="public/images/teasers/1.jpg"
                      width="100%"
                    />
                  </picture>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          aria-hidden={true}
          className="slick-slide"
          data-index={3}
          onClick={[Function]}
          style={
            Object {
              "outline": "none",
              "width": 0,
            }
          }
          tabIndex="-1"
        >
          <div>
            <div
              className="c3 ewb-slider--element"
              onClickCapture={[Function]}
              style={
                Object {
                  "display": "inline-block",
                  "width": "100%",
                }
              }
              tabIndex={-1}
            >
              <div>
                <div
                  className="c4"
                  width="100%"
                >
                  <picture
                    style={
                      Object {
                        "height": undefined,
                        "width": "100%",
                      }
                    }
                  >
                    <img
                      alt="featured-product-76257815"
                      loading="eager"
                      src="public/images/teasers/04.jpg"
                      width="100%"
                    />
                  </picture>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          aria-hidden={true}
          className="slick-slide"
          data-index={4}
          onClick={[Function]}
          style={
            Object {
              "outline": "none",
              "width": 0,
            }
          }
          tabIndex="-1"
        >
          <div>
            <div
              className="c3 ewb-slider--element"
              onClickCapture={[Function]}
              style={
                Object {
                  "display": "inline-block",
                  "width": "100%",
                }
              }
              tabIndex={-1}
            >
              <div>
                <div
                  className="c4"
                  width="100%"
                >
                  <picture
                    style={
                      Object {
                        "height": undefined,
                        "width": "100%",
                      }
                    }
                  >
                    <img
                      alt="featured-product-76257815"
                      loading="eager"
                      src="public/images/teasers/05.jpg"
                      width="100%"
                    />
                  </picture>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="c1 slick-arrow slick-next"
      data-role="none"
      onClick={[Function]}
      style={
        Object {
          "display": "block",
        }
      }
    >
      <div
        className="c5"
        color="white"
        rotate={0}
      >
        <svg
          height={30}
          viewBox="0 0 24 24"
          width={30}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.78 11.28a.75.75 0 01.22.53v.38a.77.77 0 01-.22.53l-5.14 5.13a.5.5 0 01-.71 0l-.71-.71a.49.49 0 010-.7L13.67 12 9.22 7.56a.5.5 0 010-.71l.71-.7a.5.5 0 01.71 0z"
          />
        </svg>
      </div>
    </div>
    <ul
      className="slick-dots"
      onMouseEnter={null}
      onMouseLeave={null}
      onMouseOver={null}
      style={
        Object {
          "display": "block",
        }
      }
    >
      <li
        className="slick-active"
      >
        <div
          onClick={[Function]}
        />
      </li>
      <li
        className=""
      >
        <div
          onClick={[Function]}
        />
      </li>
      <li
        className=""
      >
        <div
          onClick={[Function]}
        />
      </li>
      <li
        className=""
      >
        <div
          onClick={[Function]}
        />
      </li>
      <li
        className=""
      >
        <div
          onClick={[Function]}
        />
      </li>
    </ul>
  </div>
</div>
`;
